### PR TITLE
Add nested network performance case, support 2 different hyper-v hosts

### PR DESCRIPTION
**Pull request by LiliDeng**
NESTED-KVM-NTTTCP-DIFFERENT-L1-PUBLIC-BRIDGE
NESTED-HYPERV-NTTTCP-DIFFERENT-L1-PUBLIC-BRIDGE
NESTED-KVM-NTTTCP-DIFFERENT-L1-NAT
NESTED-KVM-LAGSCOPE-DIFFERENT-L1-PUBLIC-BRIDGE
AZURE-NESTED-KVM-NTTTCP-DIFFERENT-L1-NAT
Support 2 different hosts, L1 VM can be either Windows or Linux.

### CODE REVIEW

**Comment by chvalean** [comment]
@LiliDeng  please change the PR name to easily identify the test area introduced or to easily identify the changes in it.
Thank you!

File: XML/TestCases/NestedVmTests.xml [hunk 1]
```diff
@@ -0,0 +1,291 @@
+<TestCases>
```
[the remark this answers is not in the document]
Comment (juhlee-microsoft) [reply XML/TestCases/NestedVmTests.xml]: Also please check out other scripts in the Linux and Windows. renaming existing script is not recommended. We should check out all other script which possibly calls this module

[the remark this answers is not in the document]
Comment (LiliDeng) [reply XML/TestCases/NestedVmTests.xml]: We have Hyper-V and kvm two types nested vm, so rename kvm into vm, I already replace all places which use kvm into vm.

File: Testscripts/Linux/nat_qemu_ifup.sh [hunk 2]
```diff
@@ -0,0 +1,130 @@
+#!/bin/sh
+#
+# Copyright IBM, Corp. 2010
```
[the remark this answers is not in the document]
Comment (lizzha) [reply Testscripts/Linux/nat_qemu_ifup.sh]: This file comes from https://wiki.qemu.org/Documentation/Networking/NAT

File: Testscripts/Linux/nested_hyperv_ntttcp_different_l1_public_bridge.sh [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+while echo $1 | grep -q ^-; do
+   declare $( echo $1 | sed 's/^-//' )=$2
+   shift
+   shift
+done
+
+#
+# Constants/Globals
+#
+SERVER_IP_ADDR="192.168.4.20"
```
[the remark this answers is not in the document]
Comment (lizzha) [reply Testscripts/Linux/nested_hyperv_ntttcp_different_l1_public_bridge.sh]: I think it's OK, it doesn't matter which ip address we set

File: Testscripts/Linux/nested_hyperv_ntttcp_different_l1_public_bridge.sh [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+log_file=$logFolder/`basename "$0"`.log
+touch $log_file
+
+IP_ADDR=$SERVER_IP_ADDR
+
+if [ "$role" == "client" ]; then
+    IP_ADDR=$CLIENT_IP_ADDR
+fi
+
+start_test()
```
[the remark this answers is not in the document]
Comment (lizzha) [reply Testscripts/Linux/nested_hyperv_ntttcp_different_l1_public_bridge.sh]: yes, use Function_Name for bash code

File: XML/TestCases/NestedVmTests.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (lizzha) [reply XML/TestCases/NestedVmTests.xml]: We have expand the nested VM automation to include both nested KVM and nested Hyper-V, i.e. Linux on KVM on Hyper-V/Azure vs. Linux on Hyper-V on Hyper-V/Azure. Thus change the file name.

File: Testscripts/Linux/nested_hyperv_ntttcp_different_l1_public_bridge.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (juhlee-microsoft) [reply Testscripts/Linux/nested_hyperv_ntttcp_different_l1_public_bridge.sh]: Close the conversation and look for the change prior to merging

**Comment by LiliDeng** [comment]
@juhlee-microsoft Update code based on comments, pull latest upstream master, execute functional testing, and run regression testing by picking up a Hyper-V test case without specify testlocation, with specify one location (server). Thanks for reviewing.

**Pull request merged**